### PR TITLE
Reference file editing bug

### LINKED ISSUE #644: Cannot Edit Attached Files on Pending Booking Requests
## Expected Behavior

Files should be populated in the Document Storage on bookings edit page. They should be movable, replaceable or deletable.

## Current Behavior

Attached Files are stored in a folder and the booking request looses the information about each individual file upon creation. the reference column only contains the path to the folder instead of an array of paths to each user uploaded file.

## Possible Solution

As discussed with @KeeganWoods1, changes to the storage mechanism should be made. reference column should store all filepaths not just the path to the folder. 

- First place to look is in BookingController::store()
- BookingController::downloadReferenceFolder($folder) should be updated receive a $booking and determine the correct folder to zip up from booking instead of a foldername.
- The download link in the "my bookings" page should reflect this new route for  `downloadReferenceFolder`
- Mostly likely will require some frontend improvements to BookingForm.vue / EditBookingForm.vue as well.

## Steps to Reproduce

1. Create / Generate a booking request
2. Visit the edit page for this booking (only works if status is pending)
3. Attempt find or manipulate the attached files

## Context (Environment)

Part of the core user flow. Was also spotted by client during usability testing.

# Bug Submission Checklist
- [x] Replication steps
- [ ] Screenshots
- [ ] Tagged commits fixing the bug for traceability
- [ ] Tagged commits updating tests to cover the reported bug	
- [ ] (optional) tagged with “performance” label if performance related

### PR DESCRIPTION
**Pull request by a-drew**
##### _What existing problem does the pull request solve?_

- Download button UI are longer replacing href links... you go to route and download by giving a booking id
- Viewing files can be done by clicking links from /view or /review pages - path and filenames hashed for security
- Removing files now works by original file name, you give name we find it and delete the hash #cashmoney #sonarSucks
- Downloaded ZIPs will still contain the original filenames

##### _Demonstrate the code is solid. </br> Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI._

I'd put a gif but its 2pm so no.

##### _List all issues this PR resolves_
- closes #644
- closes #647 (replaces)

##### _Complete these items before requesting reviews_
- [X] Static Analysis Rules Validated
- [X] All commits follow naming conventions

### Sub-tasks Checklist
- [X] UI Implementation
- [X] UI Test
- [X] Backend Implementation
- [X] Backend Tests
- [ ] Integration Tests
- [ ] System Tests
- [ ] Documentation

### CODE REVIEW

**Review by joeyabou — DISMISSED** [review]
Great work this bug was part of a vital part of the application that caused a lot of problems. This pr closes exactly the issues referenced. Looks good to me!

**Review by danseremet — CHANGES_REQUESTED** [review]
I tested this locally, I mostly see the right behavior, but something strange happens when I upload and delete a file when editing the booking.
WhatsHappening

Steps:
1. Create booking with 2 files and submit.
2. Edit booking, add a new file, remove an old file, submit.
3. Try to edit booking, it does not show the whole page.

Console logs:
image

File: app/Http/Requests/CreateBookingRequest.php [hunk 1]
```diff
@@ -28,6 +28,16 @@ public function rules()
...
             'reservations.*.end_time' => ['required', 'date'],
+
+            'files' => [
+                'required_if:event.food.self_catered,true',
+                'required_if:event.alcohol,true',
+                'required_if:event.children,true',
+                'required_if:event.appliances,true',
+                'required_if:event.bake_sale,true',
+                //'mimes:pdf,doc,docx',
+                'max:50000', // 50Mb
```
Comment (alexstojda) [inline app/Http/Requests/CreateBookingRequest.php:39]: appreciate the comment

**Review by alexstojda — DISMISSED** [review]
Looks good mate

**Review by EvanDime — DISMISSED** [review]

**Review by danseremet — DISMISSED** [review]
Now it works, nice!

**Review by EvanDime — APPROVED** [review]

**Review by Alex-Hayeur — APPROVED** [review]

**Pull request merged**